content/use-cases/use-an-escrow-as-a-smart-contract.md

```diff
@@ -0,0 +1,149 @@
+
+Using an XRP Ledger escrow to provide this smart contract is a great arrangement because the party planner, as the third-party oracle, never "holds" the funds as one might in a traditional escrow arrangement, and can't possibly take the funds for themselves.
+
+Here’s a roadmap to the high-level tasks that these participants need to complete to use an escrow as a smart contract.
+
+
+{% set n = cycler(* range(1,99)) %}
+
+<span class="use-case-step-num">{{n.next()}}</span>
+## Meet the prerequisites
+
+The party host (sender) must have:
+
+- An XRP Ledger [account](accounts.html#creating-accounts) that holds enough XRP to pay for escrow and any fees incurred.
+
+- Access to a [`rippled` server](install-rippled.html) that they can use to create the escrow.
+
+The party band (receiver) must have:
+
+- An XRP Ledger [account](accounts.html#creating-accounts) that can receive the XRP paid by the escrow.
+
+- The ability to look up the details of an XRP Ledger transaction hash. For example, they can use the [RPC Tool](https://developers.ripple.com/xrp-ledger-rpc-tool.html) or access a [`rippled` server](install-rippled.html) that they can use to look up the transaction hash.
+
+The party planner (oracle) must have:
+
+- The ability to generate a condition and a fulfillment.
+
+- Access to a [`rippled` server](install-rippled.html) that they can use to submit the fulfillment value and finish the escrow.
+
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+## Define the terms of the smart contract
+
+To create the escrow as a smart contract, the participants must first define the terms of the contract. In this scenario, the participants need to agree on the following details.
+
+- Should the escrow be fulfilled after a specific time has passed or after a cryptographic condition has been fulfilled?
+
+    - *[Conditionally-held escrow](send-a-conditionally-held-escrow.html)*: In this case, the participants agree to use a conditionally-held escrow. This is a good choice because the smart contract can be fulfilled only after the party planner (oracle) verifies that the party band completed their set and submits the cryptographic condition and fulfillment to finish the escrow and release the XRP.
+
+    To handle a no-show scenario, the participants configure the escrow to be eligible for cancellation after a specific time. Typically, this cancellation time should be after the time at which the party planner is expected to finish the escrow. The the cancellation time in this way helps prevent the party host from maliciously cancelling the escrow before the party planner can finish the escrow.
```

Comment:
This isn't quite right.

An XRP Ledger Escrow cannot be "maliciously" canceled. The escrow exists in three states (I just made up the names of these states for explanation. They're not used elsewhere):

- "Held." No one can finish or cancel it.
- "Ripe." Anyone can finish it. No one can cancel it.
- "Expired." Anyone can cancel it. No one can finish it.

A time-based escrow is "ripe" after the `FinishAfter` time has passed.

A conditional escrow is "ripe" after the oracle has published the fulfillment.

A time-and-conditional escrow is "ripe" after both have happened.

An escrow that passes its `CancelAfter` time is expired, regardless of what happened with the fulfillment. An escrow that doesn't have a `CancelAfter` time never expires. If you're using an oracle, you _definitely_ want to use `CancelAfter` so that you can get your money back if the oracle does not publish the fulfillment.